config/go.d/prometheus.conf

```diff
@@ -0,0 +1,160 @@
+# netdata go.d.plugin configuration for prometheus
+#
+# This file is in YaML format. Generally the format is:
+#
+# name: value
+#
+# There are 2 sections:
+#  - GLOBAL
+#  - JOBS
+#
+#
+# [ GLOBAL ]
+# These variables set the defaults for all JOBs, however each JOB may define its own, overriding the defaults.
+#
+# The GLOBAL section format:
+# param1: value1
+# param2: value2
+#
+# Currently supported global parameters:
+#  - update_every
+#    Data collection frequency in seconds. Default: 1.
+#
+#  - autodetection_retry
+#    Re-check interval in seconds. Attempts to start the job are made once every interval.
+#    Zero means not to schedule re-check. Default: 0.
```

Comment:
I don't understand this at all. Can you explain how this works? Then maybe we can reword this and use a better name?

Reply:
This came from python plugin, same option there. 

How it works:

 - create job
 - **before run it**: try to check (module interface has `Check() bool`)
 - if check is success - run
 - if check is fail - drop the job

Sometimes you dont want to drop the job after the check and want the plugin to keep checking it. This options allows you to configure re-check interval.

This is related code from the plugin

https://github.com/ilyam8/go-orchestrator/blob/e21d211f5a17d82ee8331989cf96cee5cafd4d8e/job/build/build.go#L238-L263

If we need to retry (retry check) the job we create retry task

```go
go retryTask(ctx, m.retryCh, cfg)
```

which is wait for `autodetection_retry` and place it in the job build queue again (instead of dropping the job)

```go
func retryTask(ctx context.Context, in chan<- confgroup.Config, cfg confgroup.Config) {
	timeout := time.Second * time.Duration(cfg.AutoDetectionRetry())
	t := time.NewTimer(timeout)
	defer t.Stop()

	select {
	case <-ctx.Done():
	case <-t.C:
		select {
		case <-ctx.Done():
		case in <- cfg:
		}
	}
}
```

The whole process is https://github.com/ilyam8/go-orchestrator/blob/master/job/build/build.go

---


> Then maybe we can reword this

It was suggested by Costa, see https://github.com/netdata/netdata/issues/2757

Reply:
Cool thanks ;)